accounts-db/src/accounts_index_storage.rs

```diff
@@ -64,34 +64,21 @@ impl BgThreads {
     ) -> Self {
         // stop signal used for THIS batch of bg threads
         let local_exit = Arc::new(AtomicBool::default());
-        let handles = Some(
-            (0..threads)
-                .map(|idx| {
-                    // the first thread we start is special
-                    let can_advance_age = can_advance_age && idx == 0;
-                    let storage_ = Arc::clone(storage);
-                    let local_exit = local_exit.clone();
-                    let system_exit = exit.clone();
-                    let in_mem_ = in_mem.to_vec();
-
-                    // note that using rayon here causes us to exhaust # rayon threads and many tests running in parallel deadlock
-                    Builder::new()
-                        .name(format!("solIdxFlusher{idx:02}"))
-                        .spawn(move || {
-                            storage_.background(
-                                vec![local_exit, system_exit],
-                                in_mem_,
-                                can_advance_age,
-                            );
-                        })
-                        .unwrap()
-                })
-                .collect(),
-        );
+
+        (0..threads)
+            .map(|idx| {
+                // the first thread we start is special
+                //                let can_advance_age = can_advance_age && idx == 0;
+                let storage_ = Arc::clone(storage);
+                //                let local_exit = local_exit.clone();
+                //                let system_exit = exit.clone();
+                let in_mem_ = in_mem.to_vec();
+                storage_.foreground(in_mem_);
```

Comment:
Can you provide more context on what this change does?

The PR description mentions that there's no replacement logic right now, but it seems like this is trying to move the thread from the background to foreground?

Reply:
Basically the functionality that was former in background() is now also in foreground(), but without any of the logic around throttling and wait times. We probably don't actually need to call foreground here since nothing has happened yet, but it also should be harmless, and replacing "background" with "foreground" here makes it more obvious in the future what the change is that needs to be made.. (i.e. foreground called from somewhere in the vm's loop)

Reply:
IIUC then, `background()` happens in a background thread while `foreground()` happens in the main working thread, so we're moving this logic to the foreground.

Reply:
yes, that was the idea.. you can use either background (once) or foreground (repeatedly).